From 0c12b00185846f9ad0ba283364038a1bbe25e317 Mon Sep 17 00:00:00 2001
From: Ye Weihua <yeweihua4@huawei.com>
Date: Sat, 11 Feb 2023 17:41:03 +0800
Subject: [PATCH 056/500] [Huawei] char: imx-rngc: fix clk_disable_unprepare
 missing in error path

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188375

--------------------------------

Function clk_prepare_enable(rngc->clk) is called in imx_rngc_probe, but
clk_disable_unprepare(rngc->clk) is missing in error path. Use goto
exchange return to fix it.

Signed-off-by: Ye Weihua <yeweihua4@huawei.com>
---
 drivers/char/hw_random/imx-rngc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/char/hw_random/imx-rngc.c b/drivers/char/hw_random/imx-rngc.c
index 9b182e5bfa87..c4301efa5ceb 100644
--- a/drivers/char/hw_random/imx-rngc.c
+++ b/drivers/char/hw_random/imx-rngc.c
@@ -289,7 +289,7 @@ static int imx_rngc_probe(struct platform_device *pdev)
 			irq, imx_rngc_irq, 0, pdev->name, (void *)rngc);
 	if (ret) {
 		dev_err(rngc->dev, "Can't get interrupt working.\n");
-		return ret;
+		goto err;
 	}
 
 	if (self_test) {
-- 
2.17.1

